File: Practica de Programacion/Practica 4 Composicion y Agregacion/Ejercicio 1/1 casa.py
class Habitacion:
    def __init__(self, nombre, tamano):
        self.nombre = nombre
        self.tamano = tamano
    
    def mostrar_info(self):
        print(f"{self.nombre}: {self.tamano}m²")


class Casa: 
    def __init__(self, direccion):
        self.direccion = direccion
        self.habitaciones = []
    
    def agregar_habitacion(self, nombre, tamano):
        self.habitaciones.append(Habitacion(nombre, tamano))
    
    def mostrar_info(self):
        print(f"\nCasa en {self.direccion}:")
        [h.mostrar_info() for h in self.habitaciones]
        print(f"Total: {len(self.habitaciones)} habitaciones, {sum(h.tamano for h in self.habitaciones)}m²")


casa = Casa("Calle Principal 123")
casa.agregar_habitacion("Dormitorio", 15)
casa.agregar_habitacion("Cocina", 10)
casa.agregar_habitacion("Baño", 5)

casa.mostrar_info()

File: Practica de Programacion/Practica 4 Composicion y Agregacion/Ejercicio 1/1 casa.java
class Habitacion {
    String nombre;
    double tamano;
    
    public Habitacion(String nombre, double tamano) {
        this.nombre = nombre;
        this.tamano = tamano;
    }
    
    void mostrarInfo() {
        System.out.println(nombre + ": " + tamano + "m²");
    }
}

class Casa {
    String direccion;
    java.util.ArrayList<Habitacion> habitaciones;
    
    public Casa(String direccion) {
        this.direccion = direccion;
        this.habitaciones = new java.util.ArrayList<>();
    }
    
    void agregarHabitacion(String nombre, double tamano) {
        habitaciones.add(new Habitacion(nombre, tamano));
    }
    
    void mostrarInfo() {
        System.out.println("\nCasa en " + direccion + ":");
        for (Habitacion h : habitaciones) {
            h.mostrarInfo();
        }
        
        double totalM2 = 0;
        for (Habitacion h : habitaciones) {
            totalM2 += h.tamano;
        }
        
        System.out.println("Total: " + habitaciones.size() + " habitaciones, " + totalM2 + "m²");
    }
}

public class Main {
    public static void main(String[] args) {
        Casa casa = new Casa("Calle Principal 123");
        casa.agregarHabitacion("Dormitorio", 15);
        casa.agregarHabitacion("Cocina", 10);
        casa.agregarHabitacion("Baño", 5);
        
        casa.mostrarInfo();
    }
}